src/SensorFactory.cc

```diff
@@ -75,17 +74,22 @@ std::shared_ptr<SensorPlugin> SensorFactory::LoadSensorPlugin(
     return std::shared_ptr<SensorPlugin>();
   }
 
-  auto pluginNames = this->dataPtr->pluginLoader.LoadLibrary(fullPath);
+  auto pluginNames = this->dataPtr->pluginLoader.LoadLib(fullPath);
+
   if (pluginNames.empty())
   {
     ignerr << "Unable to load sensor plugin file for [" << fullPath << "]\n";
     return std::shared_ptr<SensorPlugin>();
   }
 
-  // Assume the first plugin is the one we're interested in
-  std::string pluginName = *(pluginNames.begin());
+  // Assume the last plugin is the one we're interested in
```

Comment:
Why the change from first to last?

Also, if keeping last, consider using a more efficient approach than iterating over all plugin names

Reply:
When you are loading a specific sensor library this may return several plugin names. For example the `unordered_set` for a gpu_lidar is: 

```bash
ignition::sensors::v4::SensorTypePlugin<ignition::sensors::v4::Lidar>
ignition::sensors::v4::SensorTypePlugin<ignition::sensors::v4::GpuLidarSensor>
```

**If we load the first plugin then we will not have lidar data**. This list should not have many elements I don't think efficiency it's a problem here.

In other cases this will be only one element for example:
 - Depth Camera Sensor:
     ```ignition::sensors::v4::SensorTypePlugin<ignition::sensors::v4::DepthCameraSensor>```
 - RgbdCameraSensor
    ```ignition::sensors::v4::SensorTypePlugin<ignition::sensors::v4::RgbdCameraSensor>```

Reply:
> If we load the first plugin then we will not have lidar data.

Wait, if it's an unordered set, then the order is meaningless. We may get either `Lidar`  - `Gpu` or `Gpu` -  `Lidar`, so we shouldn't rely on one of them being the first or last. I don't know why that's been working, maybe we've just been lucky that all compilers are returning the same thing - or maybe no one noticed that they aren't. But I think the standard doesn't offer any guarantees.

> the unordered_set for a gpu_lidar is:

Argh I see. In that specific case, `GpuLidarSensor` inherits from `Lidar`. Maybe there's something smart we can do by instantiating both and checking their relationship. But this doesn't scale well or is general enough.

---

I guess this problem was already there and is not introduced by this PR. So I guess if we find something that works for all current CI that's good enough for now :woman_shrugging: